util/configmap_util.go

```diff
@@ -0,0 +1,44 @@
+package util
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+
+	errorsutil "github.com/argoproj/argo/util/errors"
+	"github.com/argoproj/argo/util/retry"
+)
+
+// GetConfigMaps retrieves a configmap value and memoizes the result
+func GetConfigMaps(clientSet kubernetes.Interface, namespace, name, key string) (string, error) {
```

Comment:
```suggestion
func GetConfigMap(clientSet kubernetes.Interface, namespace, name, key string) (string, error) {
```

This only gets one config map?

Reply:
I would actually call this `GetConfigMapValue` as it returns the value of a specific key and not the config map as a whole